project/data_analysis/python/so/so_prepare_sim_data.py

```diff
@@ -0,0 +1,347 @@
+                                                                                    N_LF=1.,
+                                                                                    N_MF=4.,
+                                                                                    N_UHF=2.,
+                                                                                    apply_beam_correction=False,
+                                                                                    apply_kludge_correction=True)
+
+# Initialize the dictionnaries with zeros
+n_ell_t = {spec_name: ell * 0 for spec_name in spec_name_list}
+n_ell_pol = deepcopy(n_ell_t)
+
+# Create a dict storing the noise computed with the SO noise calculator
+f_pairs_LAT = [
+    "LAT_27xLAT_27",
+    "LAT_39xLAT_39",
+    "LAT_93xLAT_93",
+    "LAT_145xLAT_145",
+    "LAT_225xLAT_225",
+    "LAT_280xLAT_280",
+    "LAT_27xLAT_39",
+    "LAT_93xLAT_145",
+    "LAT_225xLAT_280"
+]
+n_ell_t_LAT_dict = {f_pair: n_ell_t_LAT[i] for i,f_pair in enumerate(f_pairs_LAT)}
+n_ell_pol_LAT_dict = {f_pair: n_ell_pol_LAT[i] for i,f_pair in enumerate(f_pairs_LAT)}
+
+# Fill the dict with non-zero noise power
+for f_pair_LAT in f_pairs_LAT:
+    if f_pair_LAT in spec_name_list:
+        n_ell_t[f_pair_LAT] = n_ell_t_LAT_dict[f_pair_LAT]
+        n_ell_pol[f_pair_LAT] = n_ell_pol_LAT_dict[f_pair_LAT]
+
+# Now let's go to Planck we will use information from the Table 4 of
+# https://arxiv.org/pdf/1807.06205.pdf
+# Planck noise will be assumed to be white for these simulations
+# we give Planck standard deviations in uk.arcmin
+f_pairs_Planck = [
+    "Planck_100xPlanck_100",
+    "Planck_143xPlanck_143",
+    "Planck_217xPlanck_217",
+    "Planck_353xPlanck_353"
+]
+
+sigma_t = {
+    "Planck_100xPlanck_100": 77.4,
+    "Planck_143xPlanck_143": 33.0,
+    "Planck_217xPlanck_217": 46.80,
+    "Planck_353xPlanck_353": 153.6
+}
+
+sigma_pol = {
+    "Planck_100xPlanck_100": 117.6,
+    "Planck_143xPlanck_143": 70.2,
+    "Planck_217xPlanck_217": 105.0,
+    "Planck_353xPlanck_353": 438.6
+}
+
+n_ell_t_Planck_dict = {}
+n_ell_pol_Planck_dict = {}
+for f_pair_Planck in f_pairs_Planck:
+    sigma_t_rad = np.deg2rad(sigma_t[f_pair_Planck]) / 60
+    sigma_pol_rad = np.deg2rad(sigma_pol[f_pair_Planck]) / 60
+
+    n_ell_t_Planck_dict[f_pair_Planck] = np.full_like(ell, sigma_t_rad ** 2)
+    n_ell_pol_Planck_dict[f_pair_Planck] = np.full_like(ell, sigma_pol_rad ** 2)
+
+# Fill the dict with non-zero noise power
+for f_pair_Planck in f_pairs_Planck:
+    if f_pair_Planck in spec_name_list:
+        n_ell_t[f_pair_Planck] = n_ell_t_Planck_dict[f_pair_Planck]
+        n_ell_pol[f_pair_Planck] = n_ell_pol_Planck_dict[f_pair_Planck]
+
+# Now let's write the n_ell_t and n_ell_pol dictionnary to disk
+# Note that we are creating a lot of small files, we could use another data format
+if d["type"] == "Dl":
+    fac = ell * (ell + 1) / (2 * np.pi)
+else:
+    fac = 1.
+
```

Comment:
could be write `fac = ell * (ell + 1) / (2 * np.pi) if d["type"] == "Dl" else 1.`